### DISCUSSION TITLE
No Cmdlet for updating Image Webpart captionText

### DISCUSSION THREAD

**Discussion by arunakshi** [pnp/powershell#2901] [Q&A]
**Use Case:** 

My original scenario was to translate text in modern pages. I was able to translate the text webparts but we image webparts with captions below it. We do not find any cmdlet to get Image webpart and get captionText. We can get image webpart CaptionText only under HtmlPropertiesData property. 

To get the image webparts, we are filtering the webparts when Title is Image, and get PropertiesJson and update the PropertiesJson property back. We cannot set HtmlPropertiesData property since this is read-only. 

We are trying the following script however the captionText is updatable. Is there any way that we can update image webpart captionText from PnP PowerShell? 

`
// my code here 

        $fromJson =  $imageControl.PropertiesJson | ConvertFrom-Json

          Add-Member -InputObject $fromJson -NotePropertyName "captionText" -NotePropertyValue $imageControl.HtmlPropertiesData.ToString()

         $toJsonData = ConvertTo-Json $fromJson
   
          Set-PnPPageWebPart -Page $newPage -Identity $imageControl.InstanceId -PropertiesJson  $toJsonData
    

`

Any suggestions would be appreciated!

Thank you.

**Comment by AndyTuke** [comment]
Did you ever find a solution for setting HtmlPropertiesData ? I'm stuck on this too trying to setup up a People web part using PnP Powershell
